numpyro/infer/initialization.py

```diff
@@ -3,121 +3,80 @@
+def init_to_median(site=None, num_samples=15):
     """
     Initialize to the prior median.
 
     :param int num_samples: number of prior points to calculate median.
     """
-    return partial(_init_to_median, num_samples=num_samples)
+    if site is None:
+        return partial(init_to_median, num_samples=num_samples)
+
+    if site['type'] == 'sample' and not site['is_observed']:
+        rng_key = site['kwargs'].get('rng_key')
+        sample_shape = site['kwargs'].get('sample_shape')
+        samples = site['fn'].sample(rng_key, sample_shape=(num_samples,) + sample_shape)
+        return np.median(samples, axis=0)
 
 
-def init_to_prior():
+def init_to_prior(site=None):
     """
     Initialize to a prior sample.
     """
-    return init_to_median(num_samples=1)
+    return init_to_median(site, num_samples=1)
 
 
-def _init_to_uniform(site, radius=2, skip_param=False):
-    if site['type'] == 'sample' and not site['is_observed']:
-        if isinstance(site['fn'], dist.TransformedDistribution):
-            fn = site['fn'].base_dist
-        else:
-            fn = site['fn']
-        value = numpyro.sample('_init', fn, sample_shape=site['kwargs']['sample_shape'])
-        base_transform = biject_to(fn.support)
-        unconstrained_value = numpyro.sample('_unconstrained_init', dist.Uniform(-radius, radius),
-                                             sample_shape=np.shape(base_transform.inv(value)))
-        return base_transform(unconstrained_value)
-
-    if site['type'] == 'param' and not skip_param:
-        # return base value of param site
-        constraint = site['kwargs'].pop('constraint', real)
-        transform = biject_to(constraint)
-        value = site['args'][0]
-        unconstrained_value = numpyro.sample('_unconstrained_init', dist.Uniform(-radius, radius),
-                                             sample_shape=np.shape(transform.inv(value)))
-        if isinstance(transform, ComposeTransform):
-            base_transform = transform.parts[0]
-        else:
-            base_transform = transform
-        return base_transform(unconstrained_value)
-
-
-def init_to_uniform(radius=2):
+def init_to_uniform(site=None, radius=2):
     """
     Initialize to a random point in the area `(-radius, radius)` of unconstrained domain.
 
     :param float radius: specifies the range to draw an initial point in the unconstrained domain.
     """
-    return partial(_init_to_uniform, radius=radius)
+    if site is None:
+        return partial(init_to_uniform, radius=radius)
 
-
-def init_to_feasible():
+    if site['type'] == 'sample' and not site['is_observed']:
+        fn = site['fn']
+        rng_key = site['kwargs'].get('rng_key')
+        sample_shape = site['kwargs'].get('sample_shape')
+        rng_key, subkey = random.split(rng_key)
+        # this is used to interpret the changes of event_shape in
+        # domain and codomain spaces
+        prototype_value = fn.sample(subkey, sample_shape=())
```

Comment:
I revise the logic here a bit: previously, we use `sample_shape` from `site['kwargs']` here but it is unnecessary. All we need is to get the shape of unconstrained space.